webapp/channels/src/components/admin_console/database/migrations_table.test.tsx

```diff
@@ -0,0 +1,124 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+import type {RenderResult} from '@testing-library/react';
+import {act, render, screen, waitFor} from '@testing-library/react';
+import {shallow} from 'enzyme';
+import React from 'react';
+import {FormattedMessage} from 'react-intl';
+
+import type {SchemaMigration} from '@mattermost/types/admin';
+
+import type {ActionResult} from 'mattermost-redux/types/actions';
+
+import {withIntl} from 'tests/helpers/intl-test-helper';
+
+import MigrationsTable from './migrations_table';
+
+const migrationsMockData: SchemaMigration[] = [];
+
+describe('components/MigrationsTable', () => {
+    const createHelpText = (
+        <FormattedMessage
+            id='admin.database.migrations_table.help_text'
+            defaultMessage='All applied migrations.'
+        />
+    );
+
+    const baseProps = {
+        createHelpText,
+        className: '',
+        actions: {
+            getAppliedSchemaMigrations: (jest.fn()).mockResolvedValue({
+                data: migrationsMockData,
+            } as ActionResult),
+        },
+    };
+
+    test('should match snapshot when there are no migrations', () => {
+        const wrapper = shallow(
+            <MigrationsTable
+                {...baseProps}
+            />);
+
+        expect(wrapper).toMatchSnapshot();
+    });
+
+    test('should have called actions.getAppliedSchemaMigrations only when first rendered', async () => {
+        let view: RenderResult;
+
+        const mockMigrations = [
+            {version: '1.0', name: 'Initial migration'},
+            {version: '1.1', name: 'Add users table'},
+        ];
+        const withDataProps = {
+            ...baseProps,
+            actions: {
+                getAppliedSchemaMigrations: (jest.fn()).mockResolvedValue({
+                    data: mockMigrations,
+                } as ActionResult),
+            },
+        };
+
+        act(() => {
+            view = render(withIntl(<MigrationsTable {...withDataProps}/>));
```

Comment:
You can use `renderWithContext` to get intl, and a few other things that might be needed.

Reply:
Updated